tests/Runner/String/Format.elm

```diff
@@ -0,0 +1,178 @@
+module Runner.String.Format exposing (format)
+
+import Diff exposing (Change(..))
+import Test.Runner.Failure exposing (InvalidReason(BadDescription), Reason(..))
+
+
+format : String -> Reason -> String
+format description reason =
+    case reason of
+        Custom ->
+            description
+
+        Equality expected actual ->
+            equalityToString { operation = description, expected = expected, actual = actual }
+
+        Comparison first second ->
+            verticalBar description first second
+
+        TODO ->
+            description
+
+        Invalid BadDescription ->
+            if description == "" then
+                "The empty string is not a valid test description."
+            else
+                "This is an invalid test description: " ++ description
+
+        Invalid _ ->
+            description
+
+        ListDiff expected actual ->
+            listDiffToString 0
+                description
+                { expected = expected
+                , actual = actual
+                }
+                { originalExpected = expected
+                , originalActual = actual
+                }
+
+        CollectionDiff { expected, actual, extra, missing } ->
+            let
+                extraStr =
+                    if List.isEmpty extra then
+                        ""
+                    else
+                        "\nThese keys are extra: "
+                            ++ (extra |> String.join ", " |> (\d -> "[ " ++ d ++ " ]"))
+
+                missingStr =
+                    if List.isEmpty missing then
+                        ""
+                    else
+                        "\nThese keys are missing: "
+                            ++ (missing |> String.join ", " |> (\d -> "[ " ++ d ++ " ]"))
+            in
+            String.join ""
+                [ verticalBar description expected actual
+                , "\n"
+                , extraStr
+                , missingStr
+                ]
+
+
+verticalBar : String -> String -> String -> String
+verticalBar comparison expected actual =
+    [ actual
+    , "╵"
+    , "│ " ++ comparison
+    , "╷"
+    , expected
+    ]
+        |> String.join "\n"
+
+
+listDiffToString :
+    Int
+    -> String
+    -> { expected : List String, actual : List String }
+    -> { originalExpected : List String, originalActual : List String }
+    -> String
+listDiffToString index description { expected, actual } originals =
```

Comment:
Yeah, I'm doing that because `format` is deprecated and will be removed in the next major release. (Runners should implement this themselves, based on what's best for their own environments, not using monospace unicode art just because there are helpers for it.)

This way in the next major release we can just delete `format` and `verticalBar` from `src` without breaking the tests.